kube-client/src/config/file_loader.rs

```diff
@@ -59,40 +59,42 @@ impl ConfigLoader {
         cluster: Option<&String>,
         user: Option<&String>,
     ) -> Result<Self, KubeconfigError> {
+        let current_context_name = &config.current_context.clone().unwrap_or_default();
         let context_name = if let Some(name) = context {
             name
-        } else if let Some(name) = &config.current_context {
-            name
         } else {
-            return Err(KubeconfigError::CurrentContextNotSet);
+            current_context_name
         };
+
         let current_context = config
             .contexts
             .iter()
-            .find(|named_context| &named_context.name == context_name)
-            .map(|named_context| &named_context.context)
+            .find(|named_context| &named_context.name.clone().unwrap_or_default() == context_name)
+            .and_then(|named_context| named_context.context.clone())
             .ok_or_else(|| KubeconfigError::LoadContext(context_name.clone()))?;
 
-        let cluster_name = cluster.unwrap_or(&current_context.cluster);
+        let current_cluster_name = &current_context.cluster.clone().unwrap_or_default();
+        let cluster_name = cluster.unwrap_or(current_cluster_name);
```

Comment:
Should be something similar to how we get the context name:

```rust
        let cluster_name = if let Some(name) = cluster {
            name
        } else if let Some(name) = &current_context.cluster {
            name
        } else {
            return Err(KubeconfigError::CurrentClusterNotSet);
        };
```

Same for `user_name` below.

Reply:
Same as the above, clusters can have empty names

Reply:
Resolving this, no longer relevant after reverting.